### PR TITLE
 Change to mask directory instead of individual mask files

### PR DESCRIPTION
**Pull request by xylar**
The name of the MOC mask file is constructed automatically from the
mesh name.

This approach is needed in anticipation of having other region masks (notably masks for individual ice shelves, see #237)

Example config files have been updated accordingly.

### CODE REVIEW

**Comment by xylar** [comment]
This needs to be merged before #237 and #246

**Comment by xylar** [comment]
I need to re-test all of the example config files to make sure they still work.  Here's the checklist:

- [ ] `anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil`
- [ ] `configs/edison/config.20170807.beta1.G_oQU240.edison`
- [ ] `configs/edison/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison`
- [ ] `configs/edison/config.20171102.beta3rc02_1850.ne30_oECv3_ICG.edison`
- [ ] `configs/edison/config.B_low_res_ice_shelves_1696_JWolfe_layout_Edison`
- [ ] `configs/lanl/config.MatchBoth_orig`
- [ ] `configs/olcf/config.20170313.beta1.A_WCYCL1850S.ne30_oECv3_ICG.edison`
- [ ] `configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison`
- [ ] `configs/olcf/config.GMPAS-IAF_oRRS18to6v3.titan`
- [ ] `configs/theta/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta`

**Comment by xylar** [comment]
@milenaveneziani, could you check the 3 configs on OLCF?  I need to get myself set up on titan and rhea but I'm not there yet.  I'd like to make sure the region mask files there have the right naming convention.  I had to fix the names on LANL IC and theta.

**Comment by milenaveneziani** [comment]
@xylar: I am testing with the config files you listed on titan. First test (18to6 run) failed because I don't have permission to read the model data (I have asked @vanroekel to change permissions when he has a chance), but I have changed the region mask filename in a way that is compatible with how this PR expects it. Second test (beta1 run) passed. Third test (beta2 run) is going, but I found a tiny typo in the corresponding config file: I will point it to you in the review section.

File: configs/edison/config.B_low_res_ice_shelves_1696_JWolfe_layout_Edison [hunk 1]
```diff
@@ -133,11 +133,8 @@ baseDirectory =  /global/project/projectdirs/acme/ACMEv0_lowres/B1850C5_ne30_v0.
 # plot on polar plot
```
Comment (milenaveneziani) [inline configs/edison/config.B_low_res_ice_shelves_1696_JWolfe_layout_Edison]: Could you add these lines in this file, right before line 46:
```
# provide an absolute path to put HTML in an alternative location (e.g. a web
 # portal)
 # htmlSubdirectory = /global/project/projectdirs/acme/www/USERNAME/RUNNAME
htmlSubdirectory = html
```

File: configs/olcf/config.20170313.beta1.A_WCYCL1850S.ne30_oECv3_ICG.edison [hunk 2]
```diff
@@ -51,7 +51,7 @@ baseDirectory = /dir/to/analysis/output
 # option:
```
Comment (milenaveneziani) [inline configs/olcf/config.20170313.beta1.A_WCYCL1850S.ne30_oECv3_ICG.edison]: Could you add these, right before line 35:
# provide an absolute path to put HTML in an alternative location (e.g. a web
 # portal)
 # htmlSubdirectory = /ccs/proj/cli115/www/USERNAME/RUNNAME
htmlSubdirectory = html
```

File: configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison [hunk 3]
```diff
@@ -139,5 +139,8 @@ baseDirectory =  /lustre/atlas/proj-shared/cli115/milena/ACMEv0_lowres/B1850C5_n
 # Supported options are Atlantic and IndoPacific
```
Comment (milenaveneziani) [inline configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison]: Could you change `mpasMeshName` to `oEC60to30v3`?

And then add, before line 54:
```
# provide an absolute path to put HTML in an alternative location (e.g. a web
# portal)
# htmlSubdirectory = /ccs/proj/cli115/www/USERNAME/RUNNAME
htmlSubdirectory = html
```

File: configs/olcf/config.GMPAS-IAF_oRRS18to6v3.titan [hunk 4]
```diff
@@ -53,7 +53,7 @@ baseDirectory = /dir/to/analysis/output
 # option:
```
Comment (milenaveneziani) [inline configs/olcf/config.GMPAS-IAF_oRRS18to6v3.titan]: This one is missing a few things (mind you, I haven't had a chance to test this yet):
- `mpasMeshName`
- there is the piece with `Directory with mapping files`, but not this newer part:
```
# Directory for region mask files
regionMaskDirectory = 
```
- these lines before line 37:
```
# provide an absolute path to put HTML in an alternative location (e.g. a web
# portal)
# htmlSubdirectory = /ccs/proj/cli115/www/USERNAME/RUNNAME
htmlSubdirectory = html
```

File: configs/theta/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta [hunk 5]
```diff
@@ -135,17 +135,9 @@ baseDirectory =  /projects/OceanClimate/mpas-analysis_data/ACMEv0_lowres/B1850C5
...
-# Mask file for ocean basin regional computation
-regionMaskFiles = /projects/OceanClimate/mpas-analysis_data/mpas_analysis/region_masks/oEC60to30v3wLI_Atlantic_region_masks_and_southern_transect.nc
-
 [regions]
 # Directory containing mask files for ocean basins and ice shelves
-regionMaskDirectory = /projects/OceanClimate/mpas-analysis_data/mpas_analysis/region_masks/
+regionMaskDirectory = /projects/OceanClimate/mpas-analysis_data/mpas_analysis/region_masks
 
 [climatologyMapSoseTemperature]
 # Times for comparison times (Jan, Feb, Mar, Apr, May, Jun, Jul, Aug, Sep, Oct,
```
Comment (milenaveneziani) [inline configs/theta/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: do we have a htmlSubdirectory on theta?

**Comment by milenaveneziani** [comment]
Test with beta2 run also passed. I am down to testing the high-res case.

**Comment by vanroekel** [comment]
@milenaveneziani, FYI that the high res data on titan is now at 
```
/lustre/atlas1/cli115/proj-shared/vanroek/run
```

File: configs/edison/config.B_low_res_ice_shelves_1696_JWolfe_layout_Edison (hunk 1 above)
Comment (xylar) [reply configs/edison/config.B_low_res_ice_shelves_1696_JWolfe_layout_Edison]: done

File: configs/olcf/config.20170313.beta1.A_WCYCL1850S.ne30_oECv3_ICG.edison (hunk 2 above)
Comment (xylar) [reply configs/olcf/config.20170313.beta1.A_WCYCL1850S.ne30_oECv3_ICG.edison]: done

File: configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison (hunk 3 above)
Comment (xylar) [reply configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison]: done

File: configs/olcf/config.GMPAS-IAF_oRRS18to6v3.titan (hunk 4 above)
Comment (xylar) [reply configs/olcf/config.GMPAS-IAF_oRRS18to6v3.titan]: I do see the `regionMaskDirectory` part in my version at the bottom.  I have changed the rest as you suggested.

File: configs/theta/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta (hunk 5 above)
Comment (xylar) [reply configs/theta/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: No, as far as I'm aware, there's no web portal on theta.

File: configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison (hunk 3 above)
Comment (milenaveneziani) [reply configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison]: I think you forgot to adjust the `mpasMeshName`.

**Comment by milenaveneziani** [comment]
@xylar: after changing the `mpasMeshName` in one of the config files, you can merge this.

Although the olcf config file for the high-res cannot be tested for the MOC right now, due to the need for `nco 4.7.0`, I have changed the MOC transect filename for the `RRS18to6` to reflect changes here, so we shouldn't have problems when the time comes.

File: configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison (hunk 3 above)
Comment (xylar) [reply configs/olcf/config.20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison]: Yes, I missed that.  I'll fix it now.

**Comment by xylar** [comment]
@milenaveneziani, I really appreciate your efforts on the RRS18to6.  We'll get it working but I agree that it shouldn't hold up this PR as long as we're pretty sure the config file will be correct once we get it working.

**Pull request merged**